File: Level-1/41.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        ArrayList<Integer> list = new ArrayList<>();
        int idx = 0;
        for(int i = 0; i < moves.length; i++){
            int n = moves[i] - 1;
            for(int j = 0; j < board.length; j++){
                if(board[j][n] != 0){
                    list.add(idx, board[j][n]);
                    if(idx > 0 && (list.get(idx) == list.get(idx - 1))){
                        list.remove(idx);
                        list.remove(idx - 1);
                        answer += 2;
                    }
                    idx = list.size();
                    board[j][n] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}

File: Level-1/41.js
function solution(board, moves) {
    let answer = 0;
    const arr = [];
    
    moves.forEach((move) => {
        for(let i = 0; i < board.length; i++) {
            const value = board[i][move - 1];
            if(value === 0) continue;
            
            if(arr.length && arr[arr.length - 1] === value) {
                answer += 2;
                arr.pop();
            }else {
                arr.push(value);
            }
            board[i][move - 1] = 0;
            break;
        }
    });
    
    return answer;
}
